impl.go

```diff
@@ -287,16 +295,23 @@ func validReceiver(recv string) bool {
 }
 
 func main() {
-	if len(os.Args) != 3 {
+	flag.Parse()
+
+	if len(flag.Args()) < 2 {
 		fmt.Fprint(os.Stderr, usage)
 		os.Exit(2)
 	}
-	recv, iface := os.Args[1], os.Args[2]
+
+	recv, iface := flag.Args()[0], flag.Args()[1]
 	if !validReceiver(recv) {
 		fatal(fmt.Sprintf("invalid receiver: %q", recv))
 	}
 
-	fns, err := funcs(iface)
+	if *sourceDirectory == "" {
+		*sourceDirectory, _ = os.Getwd()
```

Comment:
See https://github.com/josharian/impl/pull/18#discussion_r122518103. I believe you have a typo in your snippet, it should be `err == nil` instead of `err != nil`, since inside the if statement is the success path rather than failure path.